### ISSUE TITLE
[dualtor] Incorrect TTL field for tunnel traffic

### ISSUE THREAD

**Issue by lolyu** [Azure/sonic-buildimage#6652]
**Description**

For the tunnel from the standby ToR to the active ToR, the TTL is in `pipe` mode.  But the TTL field in the packet received by the server doesn't match the behavior of `pipe` mode.

**Steps to reproduce the issue:**
1. Apply mux config to both active ToR and standby ToR
```
admin@str2-7050cx3-acs-06:~$ docker exec -it swss bash
root@str2-7050cx3-acs-06:/# cat > mux.json<<EOF
> [
>     {
>         "MUX_CABLE_TABLE:Ethernet8" : {
>             "state":"active"
>         },
>         "OP": "SET"
>     }
> ]
> EOF
root@str2-7050cx3-acs-06:/# swssconfig mux.json
```
```
admin@str2-7050cx3-acs-07:~$ docker exec -it swss bash
root@str2-7050cx3-acs-07:/# cat > mux.json<<EOF
> [
>     {
>         "MUX_CABLE_TABLE:Ethernet8" : {
>             "state":"standby"
>         },
>         "OP": "SET"
>     }
> ]
> EOF
root@str2-7050cx3-acs-07:/# swssconfig mux.json
```
2. Assign ptf `eth2` with IP address `192.168.0.3/21` and add neighbor
```
root@c2431399efc7:~# ifconfig eth2
eth2: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 9216
        inet 192.168.0.3  netmask 255.255.248.0  broadcast 192.168.7.255
        ether fa:71:ae:7c:bd:02  txqueuelen 1000  (Ethernet)
        RX packets 75310  bytes 10078152 (9.6 MiB)
        RX errors 0  dropped 262  overruns 0  frame 0
        TX packets 52505  bytes 3270252 (3.1 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```
```
admin@str2-7050cx3-acs-06:~$ sudo ip -4 neigh add 192.168.0.3 lladdr fa:71:ae:7c:bd:02 dev Vlan1000
```
3. Send packet from T1 switches to the standby ToR
the original packet from T1 to standby ToR:
```
###[ Ethernet ]###
  dst       = d4:af:f7:4d:af:18
  src       = fa:71:ae:7c:bd:02
  type      = 0x800
###[ IP ]###
     version   = 4L
     ihl       = 5L
     tos       = 0x3e
     len       = 86
     id        = 1
     flags     =
     frag      = 0L
     ttl       = 50
     proto     = hopopt
     chksum    = 0xc5bc
     src       = 1.1.1.1
     dst       = 192.168.0.3
     \options   \
###[ Raw ]###
        load      = '\x00\x01\x02\x03\x04\x05\x06\x07\x08\t\n\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f !"#$%&\'()*+,-./0123456789:;<=>?@A'
```
4. dump the tunnel traffic
```
###[ Ethernet ]###
  dst       = 52:54:00:25:6e:ed
  src       = d4:af:f7:4d:af:18
  type      = 0x800
###[ IP ]###
     version   = 4L
     ihl       = 5L
     tos       = 0x2
     len       = 106
     id        = 301
     flags     =
     frag      = 0L
     ttl       = 22
     proto     = ipencap
     chksum    = 0x8f1f
     src       = 10.1.0.33
     dst       = 10.1.0.32
     \options   \
###[ IP ]###
        version   = 4L
        ihl       = 5L
        tos       = 0x3e
        len       = 86
        id        = 1
        flags     =
        frag      = 0L
        ttl       = 49
        proto     = hopopt
        chksum    = 0xc6bc
        src       = 1.1.1.1
        dst       = 192.168.0.3
        \options   \
###[ Raw ]###
           load      = '\x00\x01\x02\x03\x04\x05\x06\x07\x08\t\n\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f !"#$%&\'()*+,-./0123456789:;<=>?@A'
```
5. dump the actual packet that sends to the server(`192.168.0.3`) from the active ToR
```
###[ Ethernet ]###
  dst       = fa:71:ae:7c:bd:02
  src       = 00:aa:bb:cc:dd:ee
  type      = 0x800
###[ IP ]###
     version   = 4L
     ihl       = 5L
     tos       = 0x3e
     len       = 86
     id        = 1
     flags     =
     frag      = 0L
     ttl       = 20
     proto     = hopopt
     chksum    = 0xe3bc
     src       = 1.1.1.1
     dst       = 192.168.0.3
     \options   \
###[ Raw ]###
        load      = '\x00\x01\x02\x03\x04\x05\x06\x07\x08\t\n\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f !"#$%&\'()*+,-./0123456789:;<=>?@A'
```
**Describe the results you received:**
TTL of the packet received by the server(`192.168.0.3`) is 20.

**Describe the results you expected:**
TTL of the packet received by the server(`192.168.0.3`) should be 49 since the original packet from T1 has TTL as 50 and the tunnel TTL mode is `pipe`.

**Additional information you deem important (e.g. issue happens only occasionally):**

* the tunnel config from the active ToR
```
127.0.0.1:6379[4]> hgetall TUNNEL|MuxTunnel0
 1) "dscp_mode"
 2) "uniform"
 3) "dst_ip"
 4) "10.1.0.32"
 5) "ecn_mode"
 6) "copy_from_outer"
 7) "encap_ecn_mode"
 8) "standard"
 9) "ttl_mode"
10) "pipe"
11) "tunnel_type"
12) "IPINIP"
13) "src_ip"
14) "10.1.0.32"
```
* the tunnel config from the standby ToR
```
127.0.0.1:6379[4]> hgetall TUNNEL|MuxTunnel0
 1) "dscp_mode"
 2) "uniform"
 3) "dst_ip"
 4) "10.1.0.33"
 5) "ecn_mode"
 6) "copy_from_outer"
 7) "encap_ecn_mode"
 8) "standard"
 9) "ttl_mode"
10) "pipe"
11) "tunnel_type"
12) "IPINIP"
13) "src_ip"
14) "10.1.0.33"
```

* show version
```
admin@str2-7050cx3-acs-06:~$ show version

SONiC Software Version: SONiC.internal-20210126-0330.post-rebase
...
```

**Comment by prsunny** [comment]
Reproduced the issue with SAI 4.3.3.1-1. @gechiang is raising CSP

**Comment by gechiang** [comment]
Case CS00012058054 raised against BRCM

**Comment by prsunny** [comment]
This is fixed in latest SAI version 4.3.3.3. Please retest with latest.

**Comment by gechiang** [comment]
Validated with SAI 4.3.3.4
outer TTL observed no longer inheriting from the inner packet.  it starts with 255...
